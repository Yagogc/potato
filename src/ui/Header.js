import styled, { keyframes } from "styled-components";
import Container from "./Container";

const logoAnim = keyframes`
  0 {
    transform: rotate(-20deg);
  }
  50% {
    transform: rotate(20deg);
  }
`;

export const Header = styled.header`
  background-color: #2c242e;
  color: white;
  text-align: center;
  font-weight: bold;
  font-size: 20px;
  max-height: 50px;
  ${Container} {
    max-height: 50px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;

export const Logo = styled.img`
  display: inline-block;
  height: 30px;
  margin-right: 15px;
  animation: ${logoAnim} 5s linear infinite;
  transform-origin: bottom;
`;
