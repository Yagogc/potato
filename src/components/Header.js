import React, { Component } from "react";
import Container from "./../ui/Container";
import { Header as HeaderWrapper, Logo } from "./../ui/Header";
import logo from "../potato.svg";
class Header extends Component {
  render() {
    return (
      <HeaderWrapper>
        <Container>
          <Logo src={logo} alt="logo" /> Potato Gallery
        </Container>
      </HeaderWrapper>
    );
  }
}

export default Header;
