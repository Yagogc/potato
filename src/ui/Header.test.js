import React from "react";
import renderer from "react-test-renderer";
import { Header, HeaderLogo, HeaderTitle } from "./Header";

test("UI - Header:", () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("UI - HeaderLogo:", () => {
  const tree = renderer.create(<HeaderLogo />).toJSON();
  expect(tree).toMatchSnapshot();
});
