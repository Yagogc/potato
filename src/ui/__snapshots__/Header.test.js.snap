// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI - Header: 1`] = `
<header
  className="sc-bwzfXH hjPlhZ"
/>
`;

exports[`UI - HeaderLogo: 1`] = `
<img
  className="sc-htpNat jjwzsn"
/>
`;
